profiling.qmd

```diff
@@ -38,43 +37,121 @@ import ProgressMeter
+For the first 18 rows of the table, the parameter being held fixed is $\sigma$, as shown in the `p` column.
+In the next set of rows the parameter being held fixed will be $\beta_1$, the intercept.
+
+There are blocks of rows for the fixed-effects ($\boldsymbol{\beta}$) parameters, the variance components (on the scale of a standard deviation), and the $\boldsymbol{\theta}$ parameters that generate the covariance factor $\boldsymbol{\Lambda}_{\boldsymbol{\theta}}$.
+(At present the correlation parameters are not profiled - we may add them later but that computation is rather awkward.)
+
+```{julia}
+show(unique(pr01.tbl.p))
+```
+
+To reiterate, the first row contains the parameter estimates for this model fit to the original response values with the constraint that $\sigma=0.130088$, instead of the global estimate $\hat{\sigma}=0.139458$ in the row for which $\zeta=0.0$.
+
+The global estimates are included in every block at the row for which $\zeta=0.0$.
+
+```{julia}
+filter(r -> iszero(r.ζ), pr01.tbl)
+```
+
+The $\zeta$ column in this table is a measure of the quality of the fit from the parameters in each row, relative to the global parameter estimates, as measured by the change in the objective (negative twice the log-likelihood).
+
+The minimum value for the objective is that at the global parameter estimates.
+The change in the objective when we constrain one parameter to a particular value has approximately a $\chi^2$ distribution on 1 degree of freedom, which is the square of a standard normal distribution, $\mathcal{Z}^2$.
+We can convert this change in the quality of the fit to the scale of the standard normal distribution by taking the *signed square root*, which is the square root of the change in the objective with the sign of $\psi-\hat{\psi}$ where $\psi$ represents the parameter being profiled.
+This is the value labelled $\zeta$ in the table.
+
+To review:
+
+- Each row in the table is the result of re-fitting the original model with the parameter in the `p` column held fixed at a particular value, as shown in the column for that parameter.
+
+- The $\zeta$ column is the signed square root of the change in the objective from the global parameter estimates.
+
+- Thus in the block of rows where $\sigma$ is held fixed, the $\zeta$ values in rows for which $\sigma<\hat\sigma$ are negative and those for which $\sigma > \hat\sigma$ have positive values of $\zeta$.
+
+- Rows in which $\zeta=0.0$ are the global parameter estimates.
+
+## Profile zeta plots
+
+@fig-kb07zetabeta shows, for each of the fixed effects parameters, $\zeta$ versus the parameter value.
 
 ```{julia}
-confint(pr01)  # defaults to the 95% confidence level
+#| code-fold: true
+#| fig-cap: "ζ versus the value of the coefficient for the fixed-effects parameters in a model of response speed for the kb07 data."
+#| label: fig-kb07zetabeta
+zetaplot!(Figure(; resolution=(1200, 350)), pr01; ptyp='β')
```

Comment:
I wonder if we should update the method here to make the ideal line and the fitted line look a bit different (color? dashed? etc)

Reply:
Sounds reasonable.  I may try to look at doing this during one of my flights tomorrow/Sunday

Reply:
The change would be made in `MixedModelsMakie.jl/src/profile.jl`.  There is a call to `lines!` and to `scatter!` to draw the observed profile and a call to `ablines!` to draw the reference line.  I'm not sure how to go about setting colors in raw Makie.  Could you perhaps do the honors, @palday?